probe_src/probe_py/analysis.py

```diff
@@ -1,44 +1,46 @@
 import typing
-import networkx
+import networkx as nx
 from .parse_probe_log import ProvLog, Op, CloneOp, ExecOp, WaitOp, CLONE_THREAD
 
+# [pid, exec_epoch_no, tid, Op | str, op_index]
+Node: typing.TypeAlias = tuple[int, int, int, Op | str, int]
```

Comment:
Let's get rid of `Op | str` since we can always retrieve the op from the 4 ints and the process_prov_log. Sorry I didn't notice that earlier.

Reply:
okay! I'll declare `op = process_tree_prov_log.processes[pid].exec_epochs[exid].threads[tid].ops[op_index].data` for every iteration of the list of nodes, because we need to access `op.flags`, `op.child_process_id`, etc.

Reply:
Yeah, to me that's preferable than redundant data in the graph we return, so sounds good. Sorry to nitpick

Reply:
This is good it made adding Node to the graph simpler. Thanks a ton!